src/drt/src/pa/FlexPA.h

```diff
@@ -352,7 +352,8 @@ class FlexPinAccessPattern
   {
     cost_ = 0;
     for (auto& ap : pattern_) {
-      cost_ += ap->getCost();
+      if (ap)
+        cost_ += ap->getCost();
```

Comment:
If `pattern_` has no access points, wouldn't the vector be empty and then the loop is skipped?

Reply:
Yes, but the test is testing whether ap is nullptr or not.

Reply:
But what is the case where an ap which is nullptr would be added to the vector?

Reply:
When you have a pin (lef PORT) with no aps. So the pref ap for this pin is null. This is a new feature. Now the router wont output an error in this case anymore.

Reply:
It's not clear to me, but I'll chalk that up to a lack of understanding how APs work.

Reply:
This vector stores the preferred ap for its corresponding pin. The correspondence is by index (of this vector and the vector of pins in the frTerm). Yes pin access is very confusing in TR.